src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/MemoryMarshal.cs

```diff
@@ -382,8 +382,61 @@ public static bool TryGetArray<T>(ReadOnlyMemory<T> memory, out ArraySegment<T>
         /// <returns>An <see cref="IEnumerable{T}"/> view of the given <paramref name="memory" /></returns>
         public static IEnumerable<T> ToEnumerable<T>(ReadOnlyMemory<T> memory)
         {
-            for (int i = 0; i < memory.Length; i++)
-                yield return memory.Span[i];
+            object? obj = memory.GetObjectStartLength(out int index, out int length);
+
+            // If the memory is empty, just return an empty array as the enumerable.
+            if (length is 0 || obj is null)
+            {
+                return Array.Empty<T>();
```

Comment:
Should this be `Enumerable.Empty<T>`? It's different to `Array.Empty<T>`, and seems to be more optimised for `IEnumerable<T>` operations when using `SpeedOpt`s.

Reply:
This is corelib. It can't reference something from System.Linq.dll, and the difference isn't worth significant contortions to work around that.